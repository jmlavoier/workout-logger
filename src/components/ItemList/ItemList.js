import React from 'react';

const ItemList = ({ item }) => (
  <tr>
    <td>{item.timeSpent}</td>
    <td>{item.workoutType}</td>
    <td>{item.date}</td>
  </tr>
);

export default ItemList;