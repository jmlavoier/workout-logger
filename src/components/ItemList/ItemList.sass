@import 'css/variables'

.item-list
  &:hover
    background-color: rgba($color-blue-secondary, 0.2)  

  td
    padding: 10px

  td:nth-child(4) > div
    padding: 5px 4px 3px 4px
    cursor: pointer
    text-align: center

.odd
  background-color: rgba($color-blue-secondary, 0.1)

.even
  background-color: transparent


        