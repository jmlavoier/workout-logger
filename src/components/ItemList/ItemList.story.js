import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import ItemList from './ItemList';

const mockItem = { 
  timeSpent: '1:00',
  workoutType: 'Run',
  date: '15/04/2018',
};

storiesOf('ItemList', module)
  .add('Default', withInfo()(() => (
    <ItemList item={mockItem} />
  )));
