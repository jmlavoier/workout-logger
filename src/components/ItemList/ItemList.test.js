import React from 'react';
import { shallow, mount } from 'enzyme';

import ItemList from './Item';

const mockItem ={ 
  timeSpent: '1:00',
  workoutType: 'Run',
  date: '15/04/2018',
};

describe('<ItemList/>', () => {
  it('Should component render item', () => {
    const { timeSpent, workoutType, date } = mockItem;

    const wrapper = mount(<Item item={mockItem} />);

    expect(wrapper.find('.time-spent').text()).toBe(timeSpent);
    expect(wrapper.find('.workout-type').text()).toBe(workoutType);
    expect(wrapper.find('.date').text()).toBe(date);
  });
});

