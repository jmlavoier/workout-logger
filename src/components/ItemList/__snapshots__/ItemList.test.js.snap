// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList/> Should component render item 1`] = `
<table>
  <tbody>
    <ItemList
      item={
        Object {
          "date": "15/04/2018",
          "timeSpent": "1:00",
          "workoutType": "Run",
        }
      }
    >
      <tr>
        <td>
          1:00
        </td>
        <td>
          Run
        </td>
        <td>
          15/04/2018
        </td>
      </tr>
    </ItemList>
  </tbody>
</table>
`;
